### PR TITLE
Add errors when files has virus or are not processed

### PR DESCRIPTION
**Pull request by slawosz**
Jira: https://regulatorydelivery.atlassian.net/jira/software/projects/COSBETA/boards/24?selectedIssue=COSBETA-1488
Screenshot 2022-05-13 180406

Moves validation logic to the validator.
Also, included https://regulatorydelivery.atlassian.net/browse/COSBETA-1506:
Screenshot 2022-05-17 145530

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looks good as expected. 
Verified all possible validation errors on accept and submit page
Also, tested formulation behaviour looks good

File: cosmetics-web/app/models/component.rb [hunk 1]
```diff
@@ -135,16 +135,28 @@ def display_root_category
     get_category_name(root_category)
   end
 
+  # This should read formulation file required
+  # TODO: find where it is used
```
Comment (scruti) [inline cosmetics-web/app/models/component.rb:139]: :question: Is this a development comment or a real TODO for the future? Seems to be used by the notification `missing_information?` method, which is used for state transition logic.

File: cosmetics-web/app/models/component.rb [hunk 2]
```diff
@@ -135,16 +135,28 @@ def display_root_category
...
     if range?
       !formulation_file.attached? && range_formulas&.empty?
     elsif exact?
       !formulation_file.attached? && exact_formulas&.empty?
+    elsif predefined? && contains_poisonous_ingredients
+      !formulation_file.attached?
     else
       false
     end
   end
```
Comment (scruti) [inline cosmetics-web/app/models/component.rb:150]: Suggestion: We can simplify the method by using a guard clause:

```suggestion
  def formulation_required?
    return false if formulation_file.attached?

    (range && range_formulas&.empty?) ||
      (exact? && exact_formulas&.empty?) ||
      (predefined? && contains_poisonous_ingredients)
  end
```

File: cosmetics-web/app/models/notification.rb [hunk 3]
```diff
@@ -123,29 +123,12 @@ def add_image(image)
...
 
-  def missing_information?
-    nano_material_required? || formulation_required? || images_missing_or_not_passed_antivirus_check?
-  end
-
-  def nano_material_required?
-    missing_nano_materials.present?
-  end
+  # TODO: this might be replaced by validator
+  def missing_information?; end
```
Comment (scruti) [inline cosmetics-web/app/models/notification.rb:131]: :question: What is the purpose of this method now that is empty?

File: cosmetics-web/app/validators/accept_and_submit_validator.rb [hunk 4]
```diff
@@ -1,9 +1,43 @@
...
+      end
+    end
+  end
+
+  def validate_image_uploads(notification)
+    notification.image_uploads.each do |image_upload|
+      if image_upload.pending_antivirus_check?
+        notification.errors.add :image_uploads, "Image #{image_upload.file.filename} is still being processed"
+      end
+      if image_upload.failed_antivirus_check?
```
Comment (scruti) [inline cosmetics-web/app/validators/accept_and_submit_validator.rb:22]: :raising_hand_man: These two conditions are mutually exclusive:

```suggestion
      elsif image_upload.failed_antivirus_check?
```

File: cosmetics-web/app/validators/accept_and_submit_validator.rb [hunk 5]
```diff
@@ -1,9 +1,43 @@
...
+    notification.components.each do |component|
+      if component.formulation_file_pending_antivirus_check?
+        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} is still being processed"
+      end
+      if component.formulation_file_failed_antivirus_check?
+        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} failed antivirus check. Remove file and try again"
+      end
+      if component.formulation_required?
+        notification.errors.add :formulation_uploads, "Item #{component.name} is missing formulation file"
       end
```
Comment (scruti) [inline cosmetics-web/app/validators/accept_and_submit_validator.rb:41]: :raising_hand: mutually exclusive conditions:
```suggestion
      if component.formulation_required?
        notification.errors.add :formulation_uploads, "Item #{component.name} is missing formulation file"
      elsif component.formulation_file_pending_antivirus_check?
        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} is still being processed"
      elsif component.formulation_file_failed_antivirus_check?
        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} failed antivirus check. Remove file and try again"
      end
```

File: cosmetics-web/app/models/component.rb (hunk 1 above)
Comment (scruti) [inline cosmetics-web/app/models/component.rb:139]: From what I've found, this is only used in the code introduced for the `AcceptAndSubmitValidator`. So... time to rename and remove the TODO comment?

**Review by scruti — APPROVED** [review]

**Pull request merged**